release: Add v1.0.0-beta2 release notes (#212)
elastic/ecctl: docs/release_notes/v1.0.0-beta2.adoc

https://download.elastic.co/downloads/ecctl/1.0.0-beta2/ecctl_1.0.0-beta2_linux_64-bit.deb[ecctl_1.0.0-beta2_linux_64-bit.deb]
https://download.elastic.co/downloads/ecctl/1.0.0-beta2/ecctl_1.0.0-beta2_linux_64-bit.rpm[ecctl_1.0.0-beta2_linux_64-bit.rpm]
https://download.elastic.co/downloads/ecctl/1.0.0-beta2/ecctl_1.0.0-beta2_linux_amd64.tar.gz[ecctl_1.0.0-beta2_linux_amd64.tar.gz]

[float]
[id="{p}-release-notes-v1.0.0-beta2-whats-new"]
==== What's new

* *AppSearch instances included in default vacate operation* .App Search instances will now be moved with the default `platform allocator vacate` command. Although App Search workloads were supported in the previous release, support for vacating said instances was missed.

* *New Commands*. We added a number of new commands to implement API functionality:
+
--
* Deployment appsearch commands:

`ecctl deployment appsearch create`:: Creates an AppSearch instance.
`ecctl deployment appsearch delete`:: Deletes a previously shut down AppSearch deployment resource.
`ecctl deployment appsearch show`:: Shows the specified AppSearch deployment.
`ecctl deployment appsearch shutdown`:: Shuts down an AppSearch deployment.

> `ecctl deployment appsearch upgrade`:: Upgrades an AppSearch deployment to the Elasticsearch deployment version.

Author: This makes sense, thanks for correcting it!

* We also added a few subcommands to the platform runner command:
+

`ecctl platform runner list`::  Lists the existing platform runners
`ecctl platform runner resync`:: Resynchronizes the search index and cache for the selected runner or all
`ecctl platform runner search`:: Performs advanced runner searching
`ecctl platform runner show`:: Shows information about the specified runner
--

* *--kind flag autocompletion*`. Commands which have a `--kind` flag, now have shell autocompletion for the allowed kinds for the command.

> * *Improved documentation*. We now include the full {p} command reference with our official documentation. We also added auto-completion instructions to the docs.

Author: We should be ready to go after this PR is merged.

* *Added an `--all` flag for all `resync` commands*. Allowing to trigger a resync not only for a specific id but for all of the indexed records.

* *Changes the default polling period for the `platform allocator vacate` operation*. Defaults to 10s instead of 1s. This change should reduce the computational impact that the `allocator vacate` operation has in both the machine running the command and the target environment where the vacate is being run.

> * *Adds --max-poll-retries and --poll-frequency flags to `platform allocator vacate`*. Allowing users to configure these settings if needed.

Reviewer: ```suggestion
* *Added `--max-poll-retries` and `--poll-frequency` flags to the `platform allocator vacate` command*. If you need these settings, you can now configure them.
```

(Is there anything else you can add here, even if it's just to say what the flags do in plain English?)

Author: So these flags basically modify the behavior of the plan poller which is used to monitor a deployment's change via the pending plan. This means that `--max-poll-retries` modifies the amount of API failures to tolerate (due to 404 because the pending plan has finished or because there's been some network errors), while `--poll-frequency` controlls how often the pending plan is obtained and then streamed to the user's terminal output, which now defaults to 10s on the vacate operations.

I hope this clarifies a bit what the flags do.

> * *Adds a `--plan-history` flag to `deployment show`*. To be able to obtain the deployment plan history from a specific deployment.

Reviewer: ```suggestion
* *Added a `--plan-history` flag to the `deployment show` command*. This flag lets you obtain the deployment plan history for a specific deployment.
```

[float]
[id="{p}-release-notes-v1.0.0-beta2-bug-fixes"]
==== Bug fixes

> * `ecctl init` returned a cryptic error when the API Key could not be verified. It now returns a human-readable error which the user can act on.

Reviewer: ```suggestion
* `ecctl init` returned a cryptic error when the API Key could not be verified. It now returns a human-readable error which you can act on.
```

(We should address audiences directly, it's part of our human and friendly tone.)

* `--kind` flag was incorrectly marked as optional in the `deployment plan cancel` command.

[float]
[id="{p}-release-notes-v1.0.0-beta2-breaking-changes"]
==== Breaking changes

> * *Renamed --type flag to --kind*. We renamed the `--type` flag to `--kind` on the {p} where applicable to keep the CLI flag names closer to what the API declares. Commands affected by this change are:

Reviewer: ```suggestion
* *Renamed `--type` flag to `--kind`*. We renamed the `--type` flag to `--kind` for {p} where applicable to keep the CLI flag names closer to what the API declares. Commands affected by this change are:
```
